src/Path/Extra.hs

```diff
@@ -43,6 +44,12 @@ parseCollapsedAbsDir = parseAbsDir . collapseFilePath
 parseCollapsedAbsFile :: MonadThrow m => FilePath -> m (Path Abs File)
 parseCollapsedAbsFile = parseAbsFile . collapseFilePath
 
+-- | Combine a 'Path Abs Dir' and a relative 'FilePath' into a 'Path Abs Dir'
```

Comment:
I'm not sure about this description. It's a `m (Path Abs Dir)`, not `Path Abs Dir`? I applaud the effort to write a nice description but I don't think we should describe the type. Ignore if pedantic ;)